docs/lang/articles/deployment/tutorial.md

```diff
@@ -4,185 +4,215 @@ sidebar_position: 1
 
-Taichi supports the following argument types:
+### 1. Compile Taichi kernel in Python script
 
-- `ti.i32`
-- `ti.f32`
-- `ti.Ndarray`
+We firstly write a Python script named `app.py`, which compiles the Taichi kernel as an artifact. Save the following code to your local machine and run the program, you will obtain an archived `module.tcm` in the same directory as `app.py`.
 
-Despite integers and floating-point numbers, we have a commonly-used data container called [`Ndarray`](https://docs.taichi-lang.org/api/taichi/lang/_ndarray/#taichi.lang._ndarray.Ndarray). It's similar to an [`ndarray`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html) in NumPy, or a [`Tensor`](https://pytorch.org/docs/stable/tensors.html) in PyTorch. It can be multidimensional and is laid out continuously in memory. If you have experienced the multidimensional arrays in C++, You can treat it as a nested array type like `float[6][14]`.
+```python
+import taichi as ti
 
-As an example of its usage, the following init kernel has an Ndarray argument named x. The dtype attribute is set to ti.f32 to specify that the Ndarray stores floating-point data, and the ndim attribute is set to 1, indicating that the Ndarray has only one dimension. Upon execution, every element within the x Ndarray will be set to zero.
+ti.init(arch=ti.vulkan)
+if ti.lang.impl.current_cfg().arch != ti.vulkan:
+    raise RuntimeError("Vulkan is not available.")
 
-```python
 @ti.kernel
-def init(x: ti.types.ndarray(dtype=ti.f32, ndim=1)):
-    for i in x:
-        x[i] = 0
+def paint(pixels: ti.types.ndarray(dtype=ti.types.vector(3, ti.f32), ndim=2)):
+    for i, j in pixels:  # Parallelized over all pixels
+        pixels[i, j][0] = float(i) / pixels.shape[1]
+        pixels[i, j][1] = float(j) / pixels.shape[0]
+        pixels[i, j][2] = 0.75
+
+mod = ti.aot.Module(ti.vulkan)
+mod.add_kernel(paint)
+mod.archive("module.tcm")
 ```
 
-After initialization, in kernel `add_base`, we want to add a floating-point number `base` to those in `x` in each frame.
+Let's dive into the code example to see what happened.
+
+We initialize Taichi specifing the backend as `ti.vulkan` at the beginning. Considering that Taichi will fall back to CPU device if the target architecture is unavailable, we check if the current backend meet our requirement.
 
 ```python
-@ti.kernel
-def add_base(x: ti.types.ndarray(ndim=1), base: ti.f32):
-    for i in range(x.shape[0]):
-        x[i] += base
+ti.init(arch=ti.vulkan)
+if ti.lang.impl.current_cfg().arch != ti.vulkan:
+    raise RuntimeError("Vulkan is not available.")
```

Comment:
I'm not sure if this check is necessary tho. I thought it was a hack for the CI. Any comments? @ailzhang